serveradmin/common/static/js/serveradmin.js

```diff
@@ -11,38 +11,28 @@ $(document).ready(function() {
     window.spinner = {
         _running: 0,
         _spinner: $('#spinner'),
-        _toggle_css: function() {
-            if (this._spinner.css('display') === 'none') {
-                this._spinner.show();
-            }
-
-            if (this._spinner.hasClass('text-success')) {
-                this._spinner.removeClass('text-success');
-                this._spinner.addClass('text-secondary');
-
-                this._spinner.css('animation-play-state', 'paused');
-            }
-            else {
-                this._spinner.removeClass('text-secondary');
-                this._spinner.addClass('text-success');
-
-                this._spinner.css('animation-play-state', 'running');
-            }
-        },
         enable: function () {
-            this._running++;
-            if (this._running > 1) {
-                return;
-            }
+            console.debug('spinner enabled');
+            this._running = Date.now();
 
-            this._toggle_css();
+            this._spinner.removeClass('text-secondary');
+            this._spinner.addClass('text-success');
+            this._spinner.css('animation-play-state', 'running');
         },
         disable: function () {
-            this._running--;
```

Comment:
To avoid calling functions more often, I suggest to still use `running` or some other boolean value to check if the spinner is actually running and can/need to be disabled.
Same could work for `enable` function.

Reply:
I would like to keep it simple here. It should be in the responsibility of the developer to start and stop only one action at a time. Everything else can be a good hint to a bug (e.g. spinner keeps running).